CIP-????/README.md

```diff
@@ -0,0 +1,194 @@
+      |                              |------------------------------>|
+      |                              |                               |
+```
+
+### Requirement Levels 
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+### Payload structure
+
+The payload MUST be encoded as a JSON string. JSON strings are semi-structured data that are human-readable. So that with a straightforward decoding into text, it will be understandable for the readers. This feature also allows the users to debug it in an easy manner, for example, with the browser debugger tools.
+
+The content of the payload will be included in the protected header of the COSESign1 signature, hence the content effects directly the behavior and security of the system. The payload MUST have the following fields: 
+
+1. The field `uri` MUST contain the full path to the endpoint, where the payload will be processed.
+
+2. Sometimes, the endpoint `uri` field is not enough to determine its purpose. The user should understand perfectly the objective of the payload which he or she is signing. That's why the payload MUST contain an `action` field with a descriptive text containing the purpose of the payload. For example, if someone calls the endpoint `/users` without an action field, they can create or delete users. However, by including the action field in the payload, it not only provides additional clarity but also effectively limits the scope of the payload.
+
+3. The payload MUST include a UNIX timestamp. The `timestamp` field is used as a nonce and a mark for the payload expiration. This field is very important in case the payload is compromised.
+
+Additional fields MAY be included in the payload. Depending on the process, it could be useful to include some aditional fields in the protected header of the signature. For example, the email information should be included in the protected header in a registration request. In that way, this payload can be used only to register that specific email and it can not be tampered. 
+
+#### JSON Schema
+```JSON
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "type": "object",
+  "properties": {
+    "uri": {
+      "type": "string",
+      "format": "uri"
+    },
+    "action": {
+      "type": "string"
+    },
+    "timestamp": {
+      "type": "integer"
+    }
+  },
+  "required": ["uri", "action", "timestamp"],
+  "additionalProperties": {
+    "type": "string"
+  }
+}
+```
+#### Minimum payload:
+
+```JSON
+{
+    "uri": "http://example.com/signin",
+    "action": "Sign in",
+    "timestamp": 1673261248,
+}
+```
+#### Sign up payload example:
+```JSON
+{
+    "uri": "http://example.com/signup",
+    "action": "Sign up",
+    "timestamp": 1673261248,
+    "email": "email@example.com"
+}
+
+```
+
```

Comment:
I have a concern with this approach.
It's using the pre-existing `api.signData` method, and these fields are pretty generic and not very identifying.  
Below it is defined for the wallet to unambiguously determine `It MUST detect when the payload is formatted using this specification.`.
How does the wallet do this, and how does the wallet/dApp guard against false positive/negative detection?

Eg, What should the wallet do if it sees `"uri"` and `"action"` but not `"timestamp"`?
1. Assume it is an Authenticated WEB3 HTTP Request that is in error? ; OR
2. Assume that it is NOT an Authenticated WEB3 HTTP Request?

And what should the wallet do in either case?

Reply:
Hi @stevenj! Apologies for the delay. This functionality seamlessly integrates with the existing implementation. To detect the payload, we simply retrieve a JSON object and check for the presence of the mandatory fields: "uri", "action", and "timestamp," as you mentioned earlier. It's important to note that it should not throw an error because CIP-0030 allows signing arbitrary payloads. The crucial aspect here is that the wallet effectively communicates to the user the specific endpoint and corresponding action associated with the request.

Reply:
@stevenj this looks resolved based on last comment but feel free to re-open (we've moving this forward & would like to know if any unresolved issues)